server/mux.go

```diff
@@ -67,127 +67,137 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
 		hr.NotFound = http.StripPrefix(opts.Basepath, hr.NotFound)
 	}
 
+	EnsureViewer := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, ViewerRoleName, opts.Logger, next)
+	}
+	EnsureEditor := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, EditorRoleName, opts.Logger, next)
+	}
+	EnsureAdmin := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, AdminRoleName, opts.Logger, next)
+	}
+
 	/* Documentation */
 	router.GET("/swagger.json", Spec())
 	router.GET("/docs", Redoc("/swagger.json"))
 
 	/* API */
 	// Sources
-	router.GET("/chronograf/v1/sources", service.Sources)
-	router.POST("/chronograf/v1/sources", service.NewSource)
+	router.GET("/chronograf/v1/sources", EnsureViewer(service.Sources))
+	router.POST("/chronograf/v1/sources", EnsureEditor(service.NewSource))
 
-	router.GET("/chronograf/v1/sources/:id", service.SourcesID)
-	router.PATCH("/chronograf/v1/sources/:id", service.UpdateSource)
-	router.DELETE("/chronograf/v1/sources/:id", service.RemoveSource)
+	router.GET("/chronograf/v1/sources/:id", EnsureViewer(service.SourcesID))
+	router.PATCH("/chronograf/v1/sources/:id", EnsureEditor(service.UpdateSource))
+	router.DELETE("/chronograf/v1/sources/:id", EnsureEditor(service.RemoveSource))
 
 	// Source Proxy to Influx; Has gzip compression around the handler
-	influx := gziphandler.GzipHandler(http.HandlerFunc(service.Influx))
+	influx := gziphandler.GzipHandler(http.HandlerFunc(EnsureViewer(service.Influx)))
 	router.Handler("POST", "/chronograf/v1/sources/:id/proxy", influx)
 
 	// Write proxies line protocol write requests to InfluxDB
-	router.POST("/chronograf/v1/sources/:id/write", service.Write)
+	router.POST("/chronograf/v1/sources/:id/write", EnsureEditor(service.Write))
 
 	// Queries is used to analyze a specific queries
-	router.POST("/chronograf/v1/sources/:id/queries", service.Queries)
+	router.POST("/chronograf/v1/sources/:id/queries", EnsureEditor(service.Queries))
```

Comment:
does this POST actually create a resource? the comment makes it sound like it just "analyzes". if just analyze, i think this may be Viewer accessible. cc @goller. mind editing the comment if it's actually about creating resources?

Reply:
If this is just submitting a query, then a Viewer should be capable. If it's editing an existing query, Editor.

Reply:
this was correct. should be Viewer.